src/java.base/share/classes/java/nio/file/Path.java

```diff
@@ -257,53 +257,55 @@ public static Path of(URI uri) {
      * determined by the position of a period character ('.', U+002E FULL STOP)
      * within the file name string. If the file name element is {@code null},
      * or if the file name string does not contain a period character, or if
-     * the only period in the file name string is its first character, then
-     * the extension is {@code null}. Otherwise, the extension is the substring
-     * after the last period in the file name string. If this last period is
-     * also the last character in the file name string, then the  extension is
-     * {@linkplain String#isEmpty empty}.
+     * the only period in the file name string is its first character,  then
+     * the extension is {@linkplain String#isEmpty empty}. Otherwise, the
+     * extension is the substring starting with the last period in the file
+     * name string. If this last period is also the last character in the file
+     * name string, then the extension contains only a single period character.
+     *
+     * @apiNote
+     * The returned extension generally includes the period separator character.
+     * For example, the extension which would be returned for a path with file
+     * name "photograph.jpg" is ".jpg".
      *
      * @implSpec
      * The default implementation is equivalent for this path to:
      * <pre>{@code
      * int lastPeriod = fileName.lastIndexOf('.');
      * if (lastPeriod <= 0)
-     *     return null;
-     * return (lastPeriod == fileName.length() - 1)
-     *     ? ""
-     *     : fileName.substring(lastPeriod + 1);
+     *     return "";
+     * else if (lastPeriod == fileName.length() - 1)
+     *     return ".";
```

Comment:
I believe you could remove these two lines (277-278) - as the case where the last period is the last character in the string should now fold into the case where it is followed by some (non-period) characters.

Reply:
Since this is part of the specification, the simpler it is, the better.